src/main/java/com/bgsoftware/superiorskyblock/menu/MenuWarps.java

```diff
@@ -46,21 +47,30 @@ public void onPlayerClick(InventoryClickEvent event, IslandWarp islandWarp) {
             MenuWarpManage.openInventory(superiorPlayer, this, islandWarp);
         }
         else {
-            if(!superiorPlayer.hasBypassModeEnabled() && plugin.getSettings().chargeOnWarp > 0) {
-                if(plugin.getProviders().getBalance(superiorPlayer).compareTo(BigDecimal.valueOf(plugin.getSettings().chargeOnWarp)) < 0){
-                    Locale.NOT_ENOUGH_MONEY_TO_WARP.send(superiorPlayer);
-                    return;
-                }
-
-                plugin.getProviders().withdrawMoney(superiorPlayer, plugin.getSettings().chargeOnWarp);
-            }
 
+            simulateClick(superiorPlayer, warpCategory.getIsland(), islandWarp.getName());
             Executor.sync(() -> {
                 previousMove = false;
-                superiorPlayer.runIfOnline(Player::closeInventory);
```

Comment:
You removed this part and moved it to the simulateClick, however there's no guarantee now that previousMove=false will be called before closing the inventory, which might lead to the previous menu opening for the player.